### PR TITLE
docs: Github actions pipeline info

### PR DESCRIPTION
**Pull request by harshshah2-crest**
- Adding general information related the each Github Actions stage which can be usefull for new developers to understand the pipeline and troubleshooting failures.
- MD format: https://github.com/splunk/addonfactory-workflow-addon-release/blob/docs/github-actions/README.md
- This doc can be updated in future with more information related to stages and more detailed steps for troubleshooting.

### CODE REVIEW

**Review by bhargav-nariyani-crest — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -30,3 +30,758 @@ jobs:
...
+
+**Troubleshooting steps for failures if any:**
+
+- Tokenise the sensitive data which is shown in the failures using PSA tool's data generator  Data Generator — pytest-splunk-addon  documentation 
+
+- If you get failures in the .samples or .sample file, replace that value with a token, and add that token's replacement, relevant details
+
+**Exception file:**
+
+- `.ge_ignore` in addon root folder All the false positive can be added in this file.
```
[the remark this answers is not in the document]
Comment (harshshah2-crest) [reply README.md:90]: Updated the info in the doc

File: README.md [hunk 2]
```diff
@@ -30,3 +30,758 @@ jobs:
...
+
+
+compliance-dependencies
+=======================
+
+**Description:**
+
+- This action scans a project for third party components and reports the results. This action contains a curation file managed by Splunk Inc.
+
+**Action used:** https://github.com/oss-review-toolkit/ort
```
[the remark this answers is not in the document]
Comment (harshshah2-crest) [reply README.md]: Updated the info in the doc

File: README.md [hunk 3]
```diff
@@ -30,3 +30,758 @@ jobs:
...
+    - NOTICE_summary
+    - NOTICE_default
+    - bom.spdx.yml
+    - AsciiDoc_disclosure_document.pdf
+```
+
+compliance-copyrights
+=====================
+
+**Description**
```
[the remark this answers is not in the document]
Comment (harshshah2-crest) [reply README.md:144]: Updated the info in the doc

**Review by kkania-splunk — APPROVED** [review]

**Pull request merged**